src/main/java/com/egov/icops_integrationkerala/service/IcopsService.java

```diff
@@ -41,12 +58,28 @@ public IcopsService(AuthUtil authUtil, AuthenticationManager authenticationManag
         this.processRequestUtil = processRequestUtil;
         this.summonsUtil = summonsUtil;
         this.requestInfoGenerator = requestInfoGenerator;
+        this.util = util;
+        this.config = config;
+        this.objectMapper = objectMapper;
+        this.restTemplate = restTemplate;
     }
 
 
     public ChannelMessage sendRequestToIcops(TaskRequest taskRequest) throws Exception {
-        ProcessRequest processRequest = icopsEnrichment.getProcessRequest(taskRequest.getTask());
+
+
+        ProcessRequest processRequest = icopsEnrichment.getProcessRequest(taskRequest);
         AuthResponse authResponse = authUtil.authenticateAndGetToken();
+
+        Location location = Location.builder()
+                .latitude(taskRequest.getTask().getTaskDetails().getRespondentDetails().getLatitude())
+                .longitude(taskRequest.getTask().getTaskDetails().getRespondentDetails().getLongitude()).build();
+
+        LocationBasedJurisdiction locationBasedJurisdiction = getLocationBasedJurisdiction(authResponse,location);
+
```

Comment:
in this method getLocationBasedJurisdiction take only location as input and call auth token internally